console/services/market_app_service.py

```diff
@@ -908,65 +899,52 @@ def _patch_rainbond_app_tag(self, eid, apps):
-                            except Exception:
-                                pass
-                    apps_min_memory[app_model_version.app_id] = min_memory
-                except ValueError:
-                    apps_min_memory[app_model_version.app_id] = min_memory
+            min_memory = 0
+            try:
+                app_temp = json.loads(app_model_version.app_template)
+                for app in app_temp.get("apps"):
+                    if app.get("extend_method_map"):
+                        try:
+                            if app.get("extend_method_map").get("init_memory"):
+                                min_memory += int(app.get("extend_method_map").get("init_memory"))
+                            else:
+                                min_memory += int(app.get("extend_method_map").get("min_memory"))
+                        except Exception:
+                            pass
+                apps_min_memory[app_model_version.app_id] = min_memory
+            except ValueError:
+                pass
+            if min_memory <= apps_min_memory.get(app_model_version.app_id, 0):
+                apps_min_memory[app_model_version.app_id] = min_memory
         return apps_min_memory
 
     # patch rainbond app versions
-    def _patch_rainbond_app_versions(self, eid, apps, is_complete=None):
-        app_ids = [app.app_id for app in apps]
-        versions = rainbond_app_repo.get_rainbond_app_version_by_app_ids(eid, app_ids, is_complete)
-        if not versions:
-            return
-
-        app_with_versions = dict()
-        for version in versions:
-            if not app_with_versions.get(version.app_id):
-                app_with_versions[version.app_id] = []
-            version_info = {
-                "is_complete": version.is_complete,
-                "version": version.version,
-                "version_alias": version.version_alias,
-                "dev_status": version.dev_status,
-            }
-            # If the versions are the same, take the last version information
-            for info in app_with_versions[version.app_id]:
-                if version_info["version"] in info["version"]:
-                    info["is_complete"] = version_info["is_complete"]
-                    info["version_alias"] = version_info["version_alias"]
-                    info["dev_status"] = version_info["dev_status"]
-            if version_info not in app_with_versions[version.app_id]:
-                app_with_versions[version.app_id].append(version_info)
-
-        apps_min_memory = self._get_rainbond_app_min_memory(versions)
+    def _patch_rainbond_app_versions(self, eid, apps):
+        all_versions = []
+        for app in apps:
+            # Get a sorted list of versions, which are the release version and the normal version
+            apvs_release = rainbond_app_repo.get_rainbond_app_versions_by_id(eid, app.app_id, True, dev_status="release")
```

Comment:
Why query twice?

Reply:
In order to query, the release version and the normal version are sorted separately, and then spliced together